sw/host/opentitanlib/src/uart/console.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use regex::{Captures, Regex};
+use std::fs::File;
+use std::io::{Read, Write};
+use std::os::unix::io::AsRawFd;
+use std::time::{Duration, Instant, SystemTime};
+
+use crate::io::uart::Uart;
+use crate::util::file;
+
+#[derive(Default)]
+pub struct UartConsole {
+    pub logfile: Option<File>,
+    pub timeout: Option<Duration>,
+    pub deadline: Option<Instant>,
+    pub exit_success: Option<Regex>,
+    pub exit_failure: Option<Regex>,
+    pub timestamp: bool,
+    pub buffer: String,
+    pub newline: bool,
+}
+
+#[derive(Clone, Copy)]
+pub enum ExitStatus {
+    None,
+    CtrlC,
+    Timeout,
+    ExitSuccess,
+    ExitFailure,
+}
+
+pub trait ReadAsRawFd: Read + AsRawFd {}
```

Comment:
Would you mind commenting that this exists so you can form a vtable? That won't be obvious to people who don't know this trick.

Reply:
Done.